agent/engine/docker_client.go

```diff
@@ -474,15 +475,8 @@ func (dg *dockerGoClient) getAuthdata(image string, authData *api.RegistryAuthen
 func (dg *dockerGoClient) CreateContainer(config *docker.Config,
 	hostConfig *docker.HostConfig,
 	name string,
-	timeout time.Duration) DockerContainerMetadata {
-	// Create a context that times out after the 'timeout' duration
-	// This is defined by the const 'createContainerTimeout'. Injecting the 'timeout'
-	// makes it easier to write tests.
-	// Eventually, the context should be initialized from a parent root context
-	// instead of TODO.
-	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
-	defer cancel()
-
+	timeout time.Duration,
```

Comment:
When returning a timeout error, the timeout value is needed, how to get this value?

Reply:
If we really want to print the timeout as part of the error, we should assemble it in the caller (which already has both of these pieces of information).

Reply:
I think logging the timeout is important, because most of the timeout values are not defined by users, so they have no idea how long they've waited for the API call in the most of the time. 

But assembling the error from the caller requires at least three lines of the same code, and this same piece of code will be spread wherever the Dock client API is called, and will require more code change, what do you think?

Reply:
You're right. I looked at the calling class (which is embedded in the transition function map) and I don't think it makes sense to change that at this point.

Reply:
OK, cool, I'll make the change accordingly.

Reply:
For now, lets update the documentation for all of these methods that `timeout` is only used to generate the error and shouldn't be used for anything else. We can also add a `TODO` for changing these methods to instead accept an error type or error string as a param to use when timeout error occurs.

Reply:
OK, sounds good.